autogen/cache/cache_factory.py

```diff
@@ -40,7 +39,12 @@ def cache_factory(seed, redis_url=None, cache_path_root=".cache"):
         disk_cache = cache_factory("myseed", None)
         ```
         """
-        if RedisCache is not None and redis_url is not None:
-            return RedisCache(seed, redis_url)
+        if redis_url is not None:
+            try:
+                from .redis_cache import RedisCache
+
+                return RedisCache(seed, redis_url)
+            except ImportError:
```

Comment:
Now thinking about it, perhaps we should issue a warning here to user that we are reverting to DiskCache even when they specify `redis_url` because the package was not installed.

Reply:
I will make the change in a follow up PR